testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/client/proxy/EncodedParamsProxyTest.java

```diff
@@ -0,0 +1,114 @@
+   }
+
+   private static String generateBaseUrl() {
+      return PortProviderUtil.generateBaseUrl(EncodedParamsProxyTest.class.getSimpleName());
+   }
+
+   /**
+    * @tpTestDetails Verify "/" in "t;hawkular/f;jk-feed?query=100%-%25ig" is sent encoded
+    * @tpSince RESTEasy 3.1.4
+    */
+   @Test
+   public void testEncodeProxy() throws Exception
+   {
+      ResteasyWebTarget target = client.target(generateBaseUrl());
+      EncodedParamsProxyInterface proxy = target.proxy(EncodedParamsProxyInterface.class);
+      Response response = proxy.encode("t;hawkular/f%3Bjk-feed", "100%-%25ig", null);
+      Assert.assertEquals(200, response.getStatus());
+      String uri = response.readEntity(String.class);
+      Assert.assertEquals(generateBaseUrl() + "/test/encode/t;hawkular%2Ff%3Bjk-feed?query=100%25-%25ig", uri);
+   }
+
+   /**
+    * @tpTestDetails Verify "/" in "t;hawkular/f;jk-feed?100-%ig ?" is sent unencoded
+    * @tpSince RESTEasy 3.1.4
+    */
+   @Test
+   public void testNoencodeProxy() throws Exception
+   {
+      ResteasyWebTarget target = client.target(generateBaseUrl());
+      EncodedParamsProxyInterface proxy = target.proxy(EncodedParamsProxyInterface.class);
+      Response response = proxy.noencode("t;hawkular/f%3Bjk-feed", "100%-%25ig", null);
+      Assert.assertEquals(200, response.getStatus());
+      String uri = response.readEntity(String.class);
+      Assert.assertEquals(generateBaseUrl() + "/test/noencode/t;hawkular/f%253Bjk-feed?query=100%25-%2525ig", uri);
+   }
+
+   /**
+    * @tpTestDetails Verify matrix parameter values are sent encoded
+    * @tpSince RESTEasy 4.0.0
+    */
+   @Test
+   public void testMatrixEncodeProxy() throws Exception
+   {
+      ResteasyWebTarget target = client.target(generateBaseUrl());
+      EncodedParamsProxyInterface proxy = target.proxy(EncodedParamsProxyInterface.class);
+      Response response = proxy.encodeMatrix("t;hawkular/f%3Bjk-feed", "100%-%25ig", null);
+      Assert.assertEquals(200, response.getStatus());
+      String uri = response.readEntity(String.class);
+      Assert.assertEquals(generateBaseUrl() + "/test/encode-matrix;matrix1=t%3Bhawkular%2Ff%3Bjk-feed;matrix2=100%25-%25ig", uri);
+   }
+
+   /**
+    * @tpTestDetails Verify matrix parameter values are sent unencoded
+    * @tpSince RESTEasy 4.0.0
+    */
+   @Test
+   public void testMatrixNoencodeProxy() throws Exception
+   {
+      ResteasyWebTarget target = client.target(generateBaseUrl());
+      EncodedParamsProxyInterface proxy = target.proxy(EncodedParamsProxyInterface.class);
+      Response response = proxy.noencodeMatrix("t;hawkular/f%3Bjk-feed", "100%-%25ig", null);
+      Assert.assertEquals(200, response.getStatus());
+      String uri = response.readEntity(String.class);
+      Assert.assertEquals(generateBaseUrl() + "/test/noencode-matrix;matrix1=t%3Bhawkular%2Ff%3Bjk-feed;matrix2=100%25-%25ig", uri);
```

Comment:
@lyind , this result looks the same as the result in testMatrixEncodeProxy().

Reply:
@ronsigal I didn't change the `@Encoding` handling for `@MatrixParam` as it was working just fine where `@PathParam` and `@QueryParam` weren't working for me. To document that and to guard against future regressions, I still added the tests `testMatrixEncodeProxy()` and `testMatrixNoencodeProxy()`.

I also noticed `@MatrixParam` behaves just the same in the encoded and non-encoded case, from the proxy client's point of view. Apart from *appearing* odd or redundant in the tests I can't find anything wrong about it.

Do you have a suggestion what should be changed in the proxy client's `@MatrixParam` handling?